modules/gapi/src/streaming/vpl/vpl_cpu_accel.cpp

```diff
@@ -0,0 +1,221 @@
+#ifdef HAVE_ONEVPL
+#include <stdlib.h>
+#include <exception>
+#include <mutex> //TODO spinlock
+
+#include "streaming/vpl/vpl_cpu_accel.hpp"
+#include "streaming/vpl/vpl_utils.hpp"
+#include "streaming/vpl/surface/frame_adapter.hpp"
+#include "streaming/vpl/surface/surface.hpp"
+#include "logger.hpp"
+
+#ifdef _WIN32
+    #include <windows.h>
+    #include <sysinfoapi.h>
+#endif
+namespace cv {
+namespace gapi {
+namespace wip {
+
+
+    
+VPLCPUAccelerationPolicy::VPLCPUAccelerationPolicy() {
+    GAPI_LOG_INFO(nullptr, "created");
+}
+
+VPLCPUAccelerationPolicy::~VPLCPUAccelerationPolicy() {
+    for (auto& pair : pool_table) {
+        pair.second.clear();
+        // do not free key here: last surface will release it
+    }
+    pool_table.clear();
+    GAPI_LOG_INFO(nullptr, "destroyed");
+}
+
+void VPLCPUAccelerationPolicy::init(session_t session) {
+    (void)session;
+    //MFXVideoCORE_SetFrameAllocator(session, mfxFrameAllocator instance)
+    GAPI_LOG_INFO(nullptr, "initialize session: " << session);
+}
+
+void VPLCPUAccelerationPolicy::deinit(session_t session) {
+    (void)session;
+    GAPI_LOG_INFO(nullptr, "deinitialize session: " << session);
+}
+
+VPLCPUAccelerationPolicy::pool_key_t
+VPLCPUAccelerationPolicy::create_surface_pool(size_t pool_size, size_t surface_size_bytes,
+                                              surface_ptr_ctr_t creator) {
+    GAPI_LOG_DEBUG(nullptr, "pool size: " << pool_size << ", surface size bytes: " << surface_size_bytes);
+
+    // create empty pool
+    pool_t pool;
+    pool.reserve(pool_size);
+
+    // allocate workplace memory area
+    size_t preallocated_raw_bytes = pool_size * surface_size_bytes;
+    size_t page_size_bytes = 4 * 1024;
+    void *preallocated_pool_memory_ptr = nullptr;
+
+#ifdef _WIN32
+    SYSTEM_INFO sysInfo;
+    GetSystemInfo(&sysInfo);
+    page_size_bytes = sysInfo.dwPageSize;
+
+    GAPI_LOG_DEBUG(nullptr, "page size: " << page_size_bytes << ", preallocated_raw_bytes: " << preallocated_raw_bytes);
+    preallocated_pool_memory_ptr = _aligned_malloc(preallocated_raw_bytes, page_size_bytes);
+#endif
+
+    if (!preallocated_pool_memory_ptr) {
+        throw std::runtime_error("VPLCPUAccelerationPolicy::create_surface_pool - failed: not enough memory."
+                                 "Requested surface count: " + std::to_string(pool_size) +
+                                 ", surface bytes: " + std::to_string(surface_size_bytes));
+    }
+
+    // fill pool with surfaces
+    std::shared_ptr<void> workspace_mem_owner (preallocated_pool_memory_ptr, [] (void *ptr){
+        GAPI_LOG_INFO(nullptr, "Free workspace memory: " << ptr);
+#ifdef _WIN32
+        _aligned_free(ptr);
```

Comment:
I would double-check it with sanitizers

Reply:
Ok, i will investigate it 
Can you mention any sanitizers in Win10?